karapace/schema_reader.py

```diff
@@ -244,6 +250,10 @@ def _get_beginning_offset(self) -> int:
     def _is_ready(self) -> bool:
         if self.ready:
             return True
+
+        if self.consumer is None:
+            raise RuntimeError("Ready check was called on reader that isn't running")
```

Comment:
I think ready check should not raise an exception but instead return `False` and maybe before that log error.

Reply:
This is only called from a single place where it's guarded by `self.consumer is not None`: https://github.com/aiven/karapace/blob/85d8cc73f538d8af11aad0d2bc8681258c4aa1f0/karapace/schema_reader.py#L295-L299.

I would have tended to agree if this was a pubic method, but since it's internal, and due to how things work I'd argue it makes more sense to just fall over if this isn't true. But perhaps in that case we should unify so that we use assert here as well, that properly communicates that it's a pre-condition.